plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java

```diff
@@ -117,6 +157,51 @@ protected static URI dockerDaemonUri(final boolean isLinux, @NotNull final Map<S
         return DEFAULT_DOCKER_MACHINE_URI;
     }
 
+    protected String getDockerHostIp() {
+        return getDockerHostIp(SystemInfo.isLinux(), System.getenv());
+    }
+
+    /**
+     * Provides the IP address that is accessible from the docker container to reach the docker host
+     *
+     * @param isLinux
+     *         if System is running on Linux
+     * @param env
+     *         should contain System environment
+     * @return docker Ip address host to be used from docker container
+     */
+    protected String getDockerHostIp(final boolean isLinux, @NotNull final Map<String, String> env) {
+        if (isLinux) {
+            // search "docker0" bridge
+            Optional<InetAddress> dockerHostInetAddress = networkFinder.getIPAddress(BRIDGE_LINUX_INTERFACE_NAME);
+            if (dockerHostInetAddress.isPresent()) {
+                return dockerHostInetAddress.get().getHostAddress();
+            }
+            // return default Docker host ip address
+            return DEFAULT_LINUX_DOCKER_HOST_IP;
+        }
+
+        // on other env, Windows/Mac, search the docker machine ip and then find the bridge used
+        String host = env.get(DOCKER_HOST_PROPERTY);
+        if (host != null) {
+           Matcher matcher = HOST_REGEXP_PATTERN.matcher(host);
+            if (matcher.matches()) {
+                String dockerIpAddress = matcher.group(1);
+                Matcher ipv4Matcher = IPV4_ADDRESS_PATTERN.matcher(dockerIpAddress);
+                if (ipv4Matcher.matches()) {
+                    String subnet = ipv4Matcher.group(1).concat(".").concat(ipv4Matcher.group(2)).concat(".").concat(ipv4Matcher.group(3));
+                    // now try to find a network interface matching this
+                   Optional<InetAddress> matchingIpAddress = networkFinder.foundInetAddressMatching(subnet);
```

Comment:
and same [here](https://github.com/codenvy/che-plugins/pull/337/files#r43016078)